File: infra/src/main/java/com/beat/infra/persistence/InfraPersistenceConfig.java
package com.beat.infra.persistence;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

@Configuration(proxyBeanMethods = false)
@ComponentScan(basePackageClasses = InfraPersistenceMarker.class)
public class InfraPersistenceConfig {
}

File: infrastructure/src/main/kotlin/com/beat/infrastructure/persistence/InfraPersistenceConfig.kt
package com.beat.infrastructure.persistence

import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

@Configuration(proxyBeanMethods = false)
@ComponentScan(basePackageClasses = [InfraPersistenceMarker::class])
class InfraPersistenceConfig
